src/ProjectOrigin.Registry.Server/Startup.cs

```diff
@@ -34,10 +35,17 @@ public void ConfigureServices(IServiceCollection services)
             .BindConfiguration(OtlpOptions.Prefix)
             .ValidateDataAnnotations()
             .ValidateOnStart();
+
+        void ConfigureResource(ResourceBuilder r)
+        {
+            r.AddService("ProjectOrigin.Registry.Server",
+                serviceInstanceId: Environment.MachineName);
+        }
         var otlpOptions = _configuration.GetSection(OtlpOptions.Prefix).GetValid<OtlpOptions>();
         if (otlpOptions.Enabled)
         {
             services.AddOpenTelemetry()
+                .ConfigureResource(ConfigureResource)
```

Comment:
I would inline it, no reason for the method to be declared (yes, just saw it is like that in the wallet, I will change it.)

```suggestion

        var otlpOptions = _configuration.GetSection(OtlpOptions.Prefix).GetValid<OtlpOptions>();
        if (otlpOptions.Enabled)
        {
            services.AddOpenTelemetry()
                .ConfigureResource(r =>
                {
                    r.AddService("ProjectOrigin.WalletSystem.Server",
                    serviceInstanceId: Environment.MachineName);
                })
```

Reply:
Just commited